packages/web-react/src/components/ScrollView/ScrollViewArrows.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { isDirectionHorizontal } from '../../constants';
+import { useStyleProps } from '../../hooks';
+import { type ScrollViewDirectionType, type SpiritScrollViewArrowsProps } from '../../types';
+import { mergeStyleProps } from '../../utils';
+import { ControlButton } from '../ControlButton';
+import { Icon } from '../Icon';
+import {
+  SCROLL_VIEW_ARROW_LABEL_HORIZONTAL_END,
+  SCROLL_VIEW_ARROW_LABEL_HORIZONTAL_START,
+  SCROLL_VIEW_ARROW_LABEL_VERTICAL_END,
+  SCROLL_VIEW_ARROW_LABEL_VERTICAL_START,
+} from './constants';
+import { useScrollViewStyleProps } from './useScrollViewStyleProps';
+
+const getDefaultArrowLabels = (direction: ScrollViewDirectionType) => {
+  const isHorizontal = isDirectionHorizontal(direction);
+
+  return {
+    start: isHorizontal ? SCROLL_VIEW_ARROW_LABEL_HORIZONTAL_START : SCROLL_VIEW_ARROW_LABEL_VERTICAL_START,
+    end: isHorizontal ? SCROLL_VIEW_ARROW_LABEL_HORIZONTAL_END : SCROLL_VIEW_ARROW_LABEL_VERTICAL_END,
+  };
+};
+
+const ScrollViewArrows = (props: SpiritScrollViewArrowsProps) => {
+  const { arrowLabels, direction, scrollStep, viewportRef, ...restProps } = props;
+  const { classProps } = useScrollViewStyleProps({
+    direction,
+  });
+  const { styleProps, props: otherProps } = useStyleProps(restProps);
+  const mergedStyleProps = mergeStyleProps('button', { classProps: classProps.arrows, styleProps });
+
+  const handleScroll = (step: number) => {
+    const scrollToDirection = isDirectionHorizontal(direction) ? 'left' : 'top';
+    viewportRef.current?.scrollBy({
+      [scrollToDirection]: step,
+      behavior: 'smooth',
+    });
+  };
+
+  const defaultArrowLabels = getDefaultArrowLabels(direction);
+  const mergedArrowLabels = { ...defaultArrowLabels, ...arrowLabels };
+
+  const isHorizontal = isDirectionHorizontal(direction);
```

Comment:
move this to line ~32 and use it in the handleScroll and pass it to the getDefaultArrowLabels, we call it now 3 times.